cloudsync/runnable.py

```diff
@@ -168,6 +168,8 @@ def start(self, *, daemon=True, **kwargs):
             self.service_name = self.__class__.__name__
         if self.__shutdown:
             raise RuntimeError("Service was stopped, create a new instance to run.")
+        if self.__thread:
+            self.__thread.join(timeout=1)  # give the old thread a chance to die
```

Comment:
Would a RuntimeError in the next if block be raised in this scenario?

Reply:
if the thread is dead or if the thread dies within the one second timeout, then `self.__thread` will become falsy, and the next block won't raise. If the thread is alive, and doesn't die within the timeout, then `__thread` with be true-ish and raise, which is the intended behavior

Reply:
Ah okay cool I'm with it